src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,124 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+import duke.DukeException;
+
+/**
+ * The TaskList class encapsulates the list of tasks and its associated methods.
+ */
+public class TaskList {
+    private final ArrayList<Task> taskList;
+
+    /**
+     * Initializes an instance of TaskList that is empty.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Adds a task to the TaskList.
+     *
+     * @param task Task to be added to the task list.
+     */
+    public void addTask(Task task) {
+        taskList.add(task);
+    }
+
+    /**
+     * Returns the number of tasks in the TaskList.
+     *
+     * @return Number of tasks in the TaskList.
+     */
+    public int getNumOfTask() {
+        return taskList.size();
+    }
+
+    /**
+     * Marks the n-th task of the TaskList.
+     * If isDone is true, mark the task as done, otherwise mark as not done.
+     *
+     * @param n Index of task to-be-marked.
+     * @param isDone Indicates whether task is done or not done.
+     * @throws DukeException If n - 1 is outside the bounds of TaskList' size.
+     */
+    public void markTaskN(int n, boolean isDone) throws DukeException {
+        try {
+            this.taskList.get(n - 1).isDoneSetter(isDone);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("The number you entered is not within the valid range.");
+        }
+    }
+
+    /**
+     * Deletes the n-th task in the TaskList.
+     *
+     * @param n Index of task to be deleted.
+     * @throws DukeException If n - 1 is outside the bounds of TaskList's size.
+     */
+    public void deleteTaskN(int n) throws DukeException {
+        try {
+            this.taskList.remove(n - 1);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("The number you entered is not within the valid range.");
+        }
+    }
+
+    /**
+     * Returns the n-th task in the TaskList.
+     *
+     * @param n Index of task to be retrieved.
+     * @return n-th task in TaskList.
+     * @throws DukeException If n - 1 is outside the bounds of TaskList's size.
+     */
+    public Task getTaskN(int n) throws DukeException {
```

Comment:
Would it be better to name this method `getTaskByIndex` as N is confusing to readers what it refers to?

Reply:
That's true! I will rename the method in future commits